brave/brave-core
# Android Sync - request camera permission only before actual QR code scan

This PR moves code which requests camera permission on Android just before the actual QR code scan.



Resolves https://github.com/brave/brave-browser/issues/8399

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Perform fresh browser installation
2. Launch browser
3. Menu => Settings => Sync
4. Expected not to have permission request
5. Pressed `Scan or enter sync code`
6. Expected to see camera permission request
7. Press `Only this time`
8. Expected to see actual image in camera preview
9. Press back
10. Restart browser
11. Menu => Settings => Sync => Scan or enter sync code
12. Press `Don't allow`
13. Expected to see toast `Camera permission was not granted` and be navigated to the main sync screen
